code/game/objects/items/teleportation.dm

```diff
@@ -309,5 +309,180 @@
+	throw_range = 10
+	flags_1 = CONDUCT_1
+	inhand_icon_state = "electronic"
+	lefthand_file = 'icons/mob/inhands/misc/devices_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/misc/devices_righthand.dmi'
+	//Uses of the device left
+	var/charges = 4
+	//The maximum number of stored uses
+	var/max_charges = 4
+	///Minimum distance to teleport user forward
+	var/minimum_teleport_distance = 4
+	///Maximum distance to teleport user forward
+	var/maximum_teleport_distance = 8
+	//How far the emergency teleport checks for a safe position
+	var/parallel_teleport_distance = 3
+
+/obj/item/teleporter/Initialize(mapload)
+	. = ..()
+	START_PROCESSING(SSobj, src)
+
+/obj/item/teleporter/Destroy(force)
+	STOP_PROCESSING(SSobj, src)
+	return ..()
+
+/obj/item/teleporter/examine(mob/user)
+	. = ..()
+	. += span_notice("[src] has [charges] out of [max_charges] charges left.")
+
+/obj/item/teleporter/attack_self(mob/user)
+	. = ..()
+	if(.)
+		return
+	attempt_teleport(user, FALSE)
+	return TRUE
+
+/obj/item/teleporter/process()
+	if(prob(10) && charges < max_charges)
+		charges++
+		if(ishuman(loc))
+			var/mob/living/carbon/human/holder = loc
+			to_chat(holder, span_notice("[src] beeps."))
+		playsound(src, 'sound/machines/twobeep.ogg', 10, TRUE, extrarange = SILENCED_SOUND_EXTRARANGE, falloff_distance = 0)
+
+/obj/item/teleporter/emp_act(severity)
+	var/teleported_something = FALSE
+	if(prob(50/severity))
+		if(ishuman(loc))
+			var/mob/living/carbon/human/user = loc
+			to_chat(user, span_danger("[src] buzzes!"))
+			attempt_teleport(user, TRUE)
+		else
+			var/turf/teleport_turf = get_turf(src)
+			for(var/mob/living/mob_on_same_tile in teleport_turf)
+				if(!teleported_something)
+					teleport_turf.visible_message(span_danger("[src] activates sporadically, teleporting everyone around it!"))
+					teleported_something = TRUE
+				attempt_teleport(mob_on_same_tile, TRUE)
+			if(!teleported_something)
+				visible_message(span_danger("[src] blinks out of existence!"))
+				do_sparks(2, 1, src)
+				qdel(src)
+
+/**
+ * Tries to teleport the user forward based on random number between min/max teleport distance vars.
+ * If destination is closed turf, try to save user from gibbing via a panic teleport.
+ * Wearing bag of holding or triggering teleport via EMP removes panic teleport, higher chance of being gibbed.
+ * Mobs on same tile as destination get telefragged.
+ **/
+/obj/item/teleporter/proc/attempt_teleport(mob/user, triggered_by_emp = FALSE)
+	if(!charges && !triggered_by_emp)
+		balloon_alert(user, "recharging!")
+		return
+
+	var/turf/current_location = get_turf(user)
+	var/area/current_area = get_area(current_location)
+	if(!current_location || current_area.area_flags & NOTELEPORT || is_away_level(current_location.z) || is_centcom_level(current_location.z) || is_reserved_level(current_location.z) || !isturf(user.loc))//If turf was not found or they're on z level 2 or >7 which does not currently exist. or if user is not located on a turf
```

Comment:
Change this to a proc (`malfuctioning`) and split it into multiple calls there.

Also, your comment is both useless and inaccurate (the >2/<7 number)